crates/dyn-abi/src/dynamic/error.rs

```diff
@@ -0,0 +1,112 @@
+use crate::{DynSolType, DynSolValue, Error, Result};
+use alloc::vec::Vec;
+use alloy_primitives::{keccak256, Selector};
+
+const PANIC_SELECTOR: Selector = Selector::new([0x4e, 0x48, 0x7b, 0x71]);
+const REVERT_SELECTOR: Selector = Selector::new([0x08, 0xc3, 0x79, 0xa0]);
+
+/// A dynamic ABI error.
+///
+/// This is a representation of a Solidity error, which can be used to decode
+/// error events.
+#[derive(Debug, Clone, PartialEq)]
+pub struct DynSolError {
+    /// Error selector.
+    pub(crate) selector: Selector,
+    /// Error body types.
+    pub(crate) body: DynSolType,
```

Comment:
`Vec<DynSolType>`?

Reply:
`DynSolEvent` uses `DynSolType` for body. could change both?